### ISSUE TITLE
Use OT Tracer Resolver

### ISSUE THREAD

**Issue by dansiviter** [oracle/helidon#1384]
## Environment Details
* Helidon Version: 1.4.1
* Helidon SE or Helidon MP: Both
* JDK version: Any
* OS: Any
* Docker version (if applicable): N/A

----------

## Problem Description
Currently Helidon uses its own Tracer builder mechanism via `io.helidon.tracing.spi.TracerProvider` which appears needlessly complicated as its tightly coupled into the config Jaeger needs. Replacing this with simply using [Tracer Resolver](/opentracing-contrib/java-tracerresolver) would be far simpler and make it easier for developers to integrate.

## Steps to reproduce
N/A

**Comment by tomas-langer** [comment]
We have integration with specific tracers (Zipkin, Jaeger - and you can use any other by implementing the SPI).
There is no Jaeger specific configuration in the common module.

The reasons we have our own approach:

- We want to use Helidon config to simply add tracer configuration to Helidon configuration
- Helidon is reactive and the OpenTracing API is expecting synchronous environment, not all works as expected if you depend fully on the OpenTracing API including the `TracerResolver`
- We have a few additions to the way the tracers work (esp. for Zipkin, we needed to add a few features) - this is to correctly support propagation of traces in JAX-RS client, WebClient, DbClient, gRPC etc.

I will have a look into an option to create a Helidon tracer provider that would itself use the TracerResolver.

**Comment by tomas-langer** [comment]
Another problem I see is that this class comes from a `contrib` module and is not part of OpenTracing API.

If you want to use this, you can implement your own Helidon `TracerProvider` service loader service. This would implement a `TracerBuilder` that needs to implement just one method:
```
    @Override
    public Tracer build() {
        return TracerResolver.resolveTracer();
    }
```

This would give you the functionality you want while using Helidon.

**Comment by tomas-langer** [comment]
I have created a PR to demonstrate the solution and to offer this to the team. If the team agrees on including this, we can easily backport to 1.4 branch

**Comment by tomas-langer** [comment]
This feature was added to `master` branch and will be available as part of our 2.0 release.

**Comment by tomas-langer** [comment]
Change is now merged into both master and 1.x branch.
In MP, the tracer resolver is enabled by default. If you put `zipkin` or `jaeger` on classpath, these have higher priority so would be used instead.
For SE there is a new module that can be configured using the Helidon `Tracer` API.
